src/oas.js

```diff
@@ -137,34 +162,39 @@ function filterPathMethods(pathMatches, targetMethod) {
   const regExp = pathToRegexp(targetMethod);
   return pathMatches
     .map(p => {
-      const captures = Object.keys(p.ref).filter(r => regExp.exec(r));
+      const captures = Object.keys(p.operation).filter(r => regExp.exec(r));
 
       if (captures.length) {
         const method = captures[0];
+        p.url.method = method.toUpperCase();
+
         return {
           url: p.url,
-          ref: p.ref[method],
+          operation: p.operation[method],
         };
       }
       return undefined;
     })
     .filter(p => p);
 }
 
-function findTargetPath(pathMatches) {
+function findTargetPath(pathMatches, index) {
   let minCount = Object.keys(pathMatches[0].url.slugs).length;
-  let candidate;
+  let logOperation;
 
   for (let m = 0; m < pathMatches.length; m += 1) {
     const selection = pathMatches[m];
     const paramCount = Object.keys(selection.url.slugs).length;
     if (paramCount <= minCount) {
       minCount = paramCount;
-      candidate = selection;
+      logOperation = selection;
     }
   }
 
-  return candidate;
+  return {
+    logOperation,
+    index,
```

Comment:
What's the index here for? Seems like it'll always be 0.

Reply:
The index is passed in by our FE based on the loop it does to instantiate new instances of the `oas` class. Based on our FE filtering, we need to identify the target file where the path was found and use it for generating an `operation` instance. It defaults to 0.

Reply:
I've ripped this out and we're maintaining it in our main repo.